frontend/src/components/common/StateHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import type { ApolloError } from "@apollo/client";
+
+import ErrorState from "./info/ErrorState";
+import LoadingState from "./info/LoadingState";
+
+interface StateHandlerProps {
+  loading: boolean;
+  error?: ApolloError;
+  isEmpty?: boolean;
+  emptyState?: React.ReactNode;
+  children: React.ReactNode | React.ReactNode[];
+  fullPage?: boolean;
+}
+
+const StateHandler = ({
+  loading,
+  error,
```

Comment:
nit: if this is a boolean, we should match naming conventions with the other boolean here `isEmpty`

Reply:
removed isEmpty. gonna keep it as `loading` and `error` to be consistent with Apollo